code/go/0chain.net/blobbercore/filestore/storage.go

```diff
@@ -385,32 +385,50 @@ func (fs *FileStore) GetBlocksMerkleTreeForChallenge(allocID string,
 		return nil, nil, common.NewError("stat_error", err.Error())
 	}
 
-	totalChunks := int(math.Ceil(float64(fi.Size()) / float64(fileData.ChunkSize)))
+	numChunks := int(math.Ceil(float64(fi.Size()) / float64(fileData.ChunkSize)))
 
 	fixedMT := util.NewFixedMerkleTree(int(fileData.ChunkSize))
+	merkleChunkSize := int(fileData.ChunkSize) / 1024
 
 	bytesBuf := bytes.NewBuffer(make([]byte, 0))
-	for i := 0; i < totalChunks; i++ {
-		n, err := io.CopyN(bytesBuf, file, fileData.ChunkSize)
-		if err != nil {
-			if n == 0 && errors.Is(err, io.EOF) {
-				break
+	for chunkIndex := 0; chunkIndex < numChunks; chunkIndex++ {
+		written, err := io.CopyN(bytesBuf, file, fileData.ChunkSize)
+
+		if written > 0 {
+			dataBytes := bytesBuf.Bytes()
+
+			errWrite := fixedMT.Write(dataBytes, chunkIndex)
+			if errWrite != nil {
+				return nil, nil, common.NewError("hash_error", errWrite.Error())
 			}
-			return nil, nil, err
-		}
 
-		dataBytes := bytesBuf.Bytes()
-		err = fixedMT.Write(dataBytes, i)
-		if err != nil {
-			return nil, nil, common.NewError("buffer_write_error", err.Error())
-		}
+			if merkleChunkSize == 0 {
```

Comment:
This merkleChunkSize checking can also be moved out of the loop

Reply:
fixed